### PR TITLE
Add support for `tslint` backed by `hound-linters`

### PR DESCRIPTION
**Pull request by blackjid**
Tslint is disabled by default, alongside any other Beta linters.

To enabled it, simplify configure your `.hound.yml`:

``` yaml
tslint:
  enabled: true
  config_file: config/tslint.json
```

This is backed by hound-linters with the pull request
https://github.com/houndci/linters/pull/49

Changes:
- Implement `Config::Tslint` and `Linter::Tslint`.
- Support `*.ts`.
- Implements the `tslint_review_job`
- Adds a default configuration to `houndci/linters`

It doesn't support `ignore_files` just yet

### CODE REVIEW

**Comment by blackjid** [comment]
Any thoughts?

**Comment by gylaz** [comment]
Looks good at first glance. I'm not sure when we'll get a chance to merge this, as we focus primarily on things we use internally. This way when things break and we need to debug them, we have experience in the language to figure things out. Right now we aren't using TypeScript at thoughtbot, so this is a risk. Personally, I'm not opposed to merging this soon-ish and keeping it as alpha/beta for a while.

File: spec/models/linter/tslint_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  describe ".can_lint?" do
+    context "given a .ts file" do
+      it "returns true" do
+        result = Linter::Tslint.can_lint?("foo.ts")
+
+        expect(result).to eq true
+      end
+    end
+
+    context "given a non- typescript file" do
```
Comment (gylaz) [inline spec/models/linter/tslint_spec.rb]: space after `-`.

File: spec/models/config/tslint_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+require "spec_helper"
+require "app/models/config/base"
+require "app/models/config/tslint"
+require "app/models/config/parser"
+require "app/models/config/parser_error"
+require "app/models/config/serializer"
+require "app/models/config/json_with_comments"
+# require "json"
```
Comment (gylaz) [inline spec/models/config/tslint_spec.rb]: Remove this?

File: spec/models/config/tslint_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+describe Config::Tslint do
+  describe "#content" do
+    it "parses the configuration using JSON" do
+      raw_config = <<-EOS.strip_heredoc
```
Comment (gylaz) [inline spec/models/config/tslint_spec.rb]: How about using `<~` instead?

File: spec/models/config/tslint_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      EOS
+      commit = stubbed_commit("config/tslint.json" => raw_config)
+      config = build_config(commit)
+
+      expect(config.content).to eq("rules" => { "no-constructor-vars" => true })
+    end
+
+    context "when configuration is linter-flavored JSON format" do
+      it "parses the configuration" do
+        raw_config = <<-EOS.strip_heredoc
```
Comment (gylaz) [inline spec/models/config/tslint_spec.rb]: How about using `<~` instead?

File: spec/models/config/tslint_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        commit = stubbed_commit("config/tslint.json" => raw_config)
+        config = build_config(commit)
+
+        expect(config.content).to eq("foo" => 1, "bar" => 2)
+      end
+    end
+  end
+
+  def build_config(commit)
+    hound_config = double(
```
Comment (gylaz) [inline spec/models/config/tslint_spec.rb]: `double` -> `instance_double`

**Comment by blackjid** [comment]
thanks for the review... TIL about the new squiggly heredocs on ruby 2.3 :)
Let me know to squash the commits

File: spec/models/linter/tslint_spec.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      excluded_paths: excluded_paths,
+      serialize: content.to_s,
+    )
+    allow(Config::Tslint).to receive(:new).and_return(stubbed_tslint_config)
+
+    stubbed_tslint_config
+  end
+
+  def raw_hound_config
+    <<-EOS.strip_heredoc
```
Comment (gylaz) [inline spec/models/linter/tslint_spec.rb]: How about using `<<~` here too?

**Comment by gylaz** [comment]
One other minor comment, then looks good. We've decided to merge this and keep it as an "alpha", meaning we won't document it for now and see how it works.

Once https://github.com/houndci/linters/pull/49 is merged and deployed, this linter will be usable.

File: spec/models/linter/tslint_spec.rb (hunk 6 above)
Comment (blackjid) [inline spec/models/linter/tslint_spec.rb]: ups!

**Comment by blackjid** [comment]
@gylaz I've rebased the code and added the owner level configuration

I also updated the https://github.com/houndci/linters/pull/49 PR

**Comment by teoljungberg** [comment]
Since houndci/linters#49 has been merged, where does that leave this?

**Comment by blackjid** [comment]
@teoljungberg @gylaz any chance to merge this?? 😁

**Comment by gylaz** [comment]
Thank you @blackjid, and sorry for the wait. I've merged this as ef17a11. I estimate we'll release this live this week or the next.

**Comment by blackjid** [comment]
thank! 👍

**Pull request closed without merging**